### PR TITLE
feat: Add docs app

### PR DESCRIPTION
**Pull request by jbrundage**
@GordonSmith please review

Signed-off-by: Jaman Brundage <jbrundage372@gmail.com>

### CODE REVIEW

**Comment by jbrundage** [comment]
image

File: apps/docs/index.html [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        <script src='https://unpkg.com/@hpcc-js/dgrid@2.0.0'></script>
+        <script src='https://unpkg.com/@hpcc-js/phosphor@2.0.0'></script>
+        <script src='https://unpkg.com/@hpcc-js/graph@2.0.0'></script>
+        <script src='https://unpkg.com/@hpcc-js/form@2.0.0'></script>
+        <script src='https://unpkg.com/@hpcc-js/html@2.0.0'></script>
+        <script src='https://unpkg.com/@hpcc-js/codemirror@2.0.0'></script>
+        <script src='https://unpkg.com/@hpcc-js/composite@2.0.0'></script>
+        <script src='https://unpkg.com/@hpcc-js/map@2.0.0'></script>
+        <script src='https://unpkg.com/@hpcc-js/tree@2.0.0'></script>
+        <script src='https://unpkg.com/@hpcc-js/timeline@2.0.0'></script>
```
Comment (GordonSmith) [inline apps/docs/index.html]: The correct versions should be pulled from gallery/src-umd/config.js

File: apps/docs/index.html [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        
+        <!-- <script src="config.js"></script> -->
+        <script src="https://rawgit.com/hpcc-systems/Visualization/candidate-2.x.x/demos/gallery/src-umd/config.js"></script>
+        <script src="index.js"></script>
+        <script src="page_config.js"></script>
+        <script src="interactions.js"></script>
+        <link rel="stylesheet" href="index.css">
+        <script>
+            $(n=>{
+                new hpccjs.other.HPCCBadge().target("logo").render();
```
Comment (GordonSmith) [inline apps/docs/index.html]: <g>

**Comment by jbrundage** [comment]
@GordonSmith I have a VSCode extension (Live Sass Compiler) that transpiles SASS/SCSS to CSS on file save. Could this be something that `tsc watch` handles?

**Comment by GordonSmith** [comment]
You need to add the commands to package.json under scripts and add any dependencies you need.  I would imagine under a script called: "compile-sass" or similar.

IOW you need to work out how to do it in the command line (and what packages you need to install) and then move that into package.json.

**Comment by jbrundage** [comment]
@GordonSmith I've found that the CLI is just called "sass" and the command (from the root directory) would be:
`sass apps/docs/index.sass apps/docs/index.css`
How would I add this to something like `npm run build`?

**Comment by jbrundage** [comment]
One thing worth mentioning is that the left menu is built using the same tree data structure that the `@hpcc-js/tree` module consumes.

The `tree_to_list` function takes two arguments. If it was written in TypeScript it would look something like: `tree_to_list(data: TreeDataShape, $container: jQueryElementSelection)`

**Comment by GordonSmith** [comment]
@jbrundage Do you have an url I can see it in?

**Comment by jbrundage** [comment]
If you hold the CTRL key as you click the icons on the far left you can view the left navigation data in the various tree widgets (click it several times to cycle through the different tree widgets)

**Comment by GordonSmith** [comment]
Thx - 2 quick comments:
1. The indentation is off for files (in red)
2. Clicking on a "folder" on the LHS should show a grid of all the items from that point down on the RHS (highlighter yellow)

image

**Pull request merged**